site/source/docs/api_reference/vr.h.rst

```diff
@@ -0,0 +1,378 @@
+.. _vr-h:
+
+====
+vr.h
+====
+
+The C APIs in `vr.h <https://github.com/kripken/emscripten/blob/master/system/include/emscripten/vr.h>`_ provide basic interfaces for interacting with WebVR from Emscripten.
+
+.. contents:: Table of Contents
+	:local:
+	:depth: 1
+
+.. _test-example-code-vr-api:
+
+Test/Example code
+-----------------
+
+The vr test code demonstrates how to use this API:
+
+	- `test_vr.c <https://github.com/kripken/emscripten/blob/master/tests/test_vr.c>`_
+
+.. _functions-vr-api:
+
+Functions
+---------
+
+Initialization
+==============
+
+.. c:function:: int emscripten_vr_init()
+
+	Initialize the emscripten VR API. This will `navigator.getVRDisplays()
+	<https://w3c.github.io/webvr/spec/1.1/#navigator-getvrdisplays-attribute>`_
+	and when completed, set the return of :c:func:`emscripten_vr_ready` to be `true`.
+
+	:returns: `1` on success, `0` if the browsers WebVR support is insufficient.
+	:rtype: int
+
+.. tip:: This call succeeding is not sufficient for use of the rest of the API. Please
+	make sure to wait until the return value of :c:func:`emscripten_vr_ready` is true.
+
+.. c:function:: int emscripten_vr_ready()
+
+	Check whether the VR API has finished initializing VR displays.
+	See also :c:func:`emscripten_vr_init`.
+
+	This function may return `1` event if WebVR is not supported in the
+	running browser.
+
+	:returns: `1` if ready, `0` otherwise.
+	:rtype: int
+
+API Queries
+===========
+
+All of the following functions require :c:func:`emscripten_vr_init` to have been
+called but do not require VR displays and can therefore be called before the return
+value of :c:func:`emscripten_vr_ready` is `true`.
+
+.. c:function:: int emscripten_vr_version_minor()
+
+	Minor version of the WebVR API currently supported by the browser.
+
+	:returns: minor version of WebVR, or `-1` if not supported or API not initialized.
```

Comment:
Here I was looking what would happen if one deinitializes the API, and if the major and minor version numbers would return -1 after deinitialization, but then noticed that deinitialization paths are not supported at all. There's a couple of reasons why we will at some point need a graceful deinit path as well:

- some pages may want to host multiple apps, or run a specific app multiple times, and they may want to depend on deinitializing the previous execution after being done, to make room for the next run,
- some apps might only run VR in certain cases, and toggle VR on/off multiple times during run,
- multithreaded apps - we'll need to define how vr initialization works with respect to multiple threads. Will emscripten_vr_init() be process-wide? or thread-specific? If thread-specific, can only one thread at a time have VR initialized, or can multiple threads initialize, and then each will have their own instance running?
- once WebVR 2.0 lands, we may need to be able to initialize both 1.1 or 2.0 version (not sure how this will pan out)

Just throwing these kind of questions on the table, for further reference and ideas, I don't expect we'll need to rush into any additions here.

Reply:
I need to implement `emscripten_vr_deinit()` for freeing the displayNames anyway. So, how would you prefer the version being handled? Set it back to -1/-1?